src/client/elm/Transaction/Event.elm

```diff
@@ -552,6 +569,54 @@ protocolUpdateDecoder =
         |> required "specificationAuxiliaryData" D.string
 
 
+decoderFromMaybe : String -> Maybe a -> D.Decoder a
+decoderFromMaybe error m =
+    case m of
+        Just n ->
+            D.succeed n
+
+        Nothing ->
+            D.fail error
+
+
+hexDecoder : D.Decoder B.Bytes
+hexDecoder =
+    D.string |> D.andThen (\str -> decoderFromMaybe "Failed to decode Hex string" (Hex.Convert.toBytes str))
+
+
+cborDecoder : Cbor.Decode.Decoder a -> B.Bytes -> D.Decoder a
+cborDecoder decoder bytes =
+    decoderFromMaybe "Failed to decode Cbor" <| Cbor.Decode.decode decoder bytes
+
+
+memoDecoder : D.Decoder Memo
+memoDecoder =
+    D.oneOf
+        [ hexDecoder
+            |> D.andThen
+                (\memoBytes ->
+                    D.oneOf
+                        [ cborDecoder Cbor.Decode.string memoBytes |> D.map MemoString
+                        , cborDecoder Cbor.Decode.int memoBytes |> D.map MemoInt
+                        ]
+                )
+        , D.string |> D.map MemoRaw
+        ]
+
+
+memoToString : Memo -> String
```

Comment:
It would be useful to document the design rationale behind this function, since it is not canonical.